api/inventorypb/agents_test.go

```diff
@@ -0,0 +1,22 @@
+package inventorypb
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// This test ensures that the AgentTypeNames map that holds the human readable agent type
+// names is up to date with the types defined in AgentType_name by the proto defenition
+// by calling the AgentTypeName function using the values from protobuf and it expects the
+// result is a non-empty string, meaning that the AgentTypeNames list matches the proto
+// definitions
+func TestAgentTypes(t *testing.T) {
+	for _, val := range AgentType_name {
+		if strings.HasSuffix(val, "INVALID") {
+			continue
+		}
+		assert.NotEmpty(t, AgentTypeName(val))
+	}
```

Comment:
We should also check the other way around – that `agentTypeNames` does not contain anything extra.

Reply:
The only problem with a reverse test is that now we have a new `types` packages having all constants but tests remain in the `inventorypb`  package to avoid dependencies.
Since `agentTypeNames` in `types` package is unexported, we cannot reference it from the `inventorypb` pkg.
Prior to the `types` package it was possible without adding extra dependencies.